src/DatePicker/component/index.ios.js

```diff
@@ -0,0 +1,137 @@
+/* @flow */
+
+import * as React from 'react';
+import {
+  View,
+  DatePickerIOS,
+  Modal,
+  TouchableWithoutFeedback,
+} from 'react-native';
+import { defaultTokens } from '@kiwicom/orbit-design-tokens';
+
+import Text from '../../Text';
+import Touchable from '../../Button/Touchable';
+import StyleSheet from '../../PlatformStyleSheet';
+
+import type { Props } from './DatePickerTypes';
+
+type State = {
+  date: Date,
+};
+
+export default class iOSDatePicker extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      date: props.date || new Date(),
+    };
+  }
+
+  componentWillReceiveProps(nextProps: Props) {
```

Comment:
Hopefully this component will be replaced until this method will be deprecated in version 17.
But just to be sure, can we at least name it `UNSAFE_componentWillReceiveProps` or use `getDerivedStateFromProps` ?

Reply:
I will refactor it to use `getDerivedStateFromProps`